### PR TITLE
frontend: Autofocuses on newly added tag input

### PR DESCRIPTION
**Pull request by amrutac**
Addresses https://github.com/coreos/tectonic-installer/issues/354
screen shot 2017-05-25 at 11 32 15 am

### CODE REVIEW

File: installer/frontend/components/aws-tags.jsx [hunk 1]
```diff
@@ -69,11 +69,11 @@ export const tagsFields = new FieldList(AWS_TAGS, {
   validator: validators.AWSTagUniqueKeys,
 });
 
-const Tag = ({row, remove, placeholder}) =>
+const Tag = ({row, remove, placeholder, autoFocus}) =>
   <div className="row" style={{padding: '0 0 20px 0'}}>
     <div className="col-xs-5" style={{paddingRight: 0}}>
       <Connect field={row.key}>
-        <Input placeholder="e.g. Name" blurry />
+        <Input placeholder="e.g. Name" blurry autoFocus={autoFocus ? autoFocus : false}/>
```
Comment (ggreer) [inline installer/frontend/components/aws-tags.jsx]: Minor thing, but I think you can get the same result with `autoFocus={!!autoFocus}` (coercing it to bool).

File: installer/frontend/components/ui.jsx [hunk 2]
```diff
@@ -681,3 +681,25 @@ export class AsyncSelect extends React.Component {
...
+class InnerFieldList_ extends React.Component {
+  render() {
+    const {value, removeField, children, fields, id} = this.props;
+    const onlyChild = React.Children.only(children);
+    const newChildren = _.map(value, (unused, i) => {
+      const row = {};
+      _.keys(fields).forEach(k => row[k] = `${id}.${i}.${k}`);
+      const childProps = { row, i, key: i, remove: () => removeField(id, i) };
+      if (i === value.length - 1) {
+        childProps.autoFocus = true;
```
Comment (ggreer) [inline installer/frontend/components/ui.jsx]: 👍

**Comment by ggreer** [comment]
The code looks fine, but there's a UX issue: If you go through the installer and enter your AWS creds and click "next", the cluster info page focuses on the tag. It should focus on the cluster name.

**Comment by amrutac** [comment]
Oh yeah, I was testing this with other fields pre-filled, looking into this.

**Comment by amrutac** [comment]
@ggreer: Updated code to autofocus only on clicking the `Add tag` button

File: installer/frontend/components/ui.jsx [hunk 3]
```diff
@@ -681,3 +681,25 @@ export class AsyncSelect extends React.Component {
...
+
+class InnerFieldList_ extends React.Component {
+  render() {
+    const {value, removeField, children, fields, id} = this.props;
+    const onlyChild = React.Children.only(children);
+    const newChildren = _.map(value, (unused, i) => {
+      const row = {};
+      _.keys(fields).forEach(k => row[k] = `${id}.${i}.${k}`);
+      const childProps = { row, i, key: i, remove: () => removeField(id, i) };
+      if (value.length > 1 && i === value.length - 1) {
```
Comment (ggreer) [inline installer/frontend/components/ui.jsx]: Why check if length is > 1? You can have 0 tags and click "add more":
screen shot 2017-05-25 at 13 37 56

Comment (amrutac) [inline installer/frontend/components/ui.jsx]: Fixed

**Pull request merged**